tests/CppWinRTAuthoringTests.cpp

```diff
@@ -173,6 +175,71 @@ TEST_CASE("CppWinRTAuthoringTests::InStruct", "[property]")
     REQUIRE(test.Prop2() == 33);
 }
 
+// Simple mock of GetValue & SetValue for testing dependency properties.
+struct mock_dependency_object
+{
+    winrt::Windows::Foundation::IInspectable GetValue(winrt::Windows::UI::Xaml::DependencyProperty const& dp) const
+    {
+        return m_values.at(dp);
+    }
+    void SetValue(winrt::Windows::UI::Xaml::DependencyProperty dp, winrt::Windows::Foundation::IInspectable const& value) const
+    {
+        // We have to do a const_cast because ordinarily SetValue is const---
+        // it's delegated to WinRT, and C++/WinRT marks most of its functions
+        // const.

```

Comment:
this seems odd... SetValue's job is to change a property on the object so the method should not be const...
if it must be, maybe changing m_values to be `mutable` instead of the const_cast?

Reply:
This is cause of the macro itself using const for the property setter. This wouldn't be needed if it weren't.